plugins/bookmarks/BookmarksSearchEngine.ts

```diff
@@ -49,21 +50,59 @@ export default class ChromiumBookmarksSearchEngine implements ISearchEngine {
 		}))
 	}
 
+	// TODO(davy): support macos
 	async openBookmark({ url }: Bookmark) {
-		exec(`xdg-open ${url}`)
+		switch (this.api.environment.platform) {
+		case 'linux':
+			spawn(`xdg-open ${url}`)
+			break
+		case 'windows': {
+			const encodedCommand = Buffer.from([
+				'Start',
+				`"${url}"`,
+			].join(' '), 'utf16le').toString('base64')
+			const powershellArgs = [
+				'-NoProfile',
+				'-NonInteractive',
+				'-WindowStyle', 'Hidden',
+				'–ExecutionPolicy', 'Bypass',
+				'-EncodedCommand',
+				encodedCommand,
+			]
+
+			spawn('powershell', {
+				argv: powershellArgs,
+			})
+			break
+		}
+		default:
+			break
+		}
 	}
 
 	private loadBookmarks() {
 		console.log('loading bookmarks')
-		const supportedBrowsers = [ 'microsoft-edge-dev', 'chromium', 'google-chrome' ]
 		const bookmarkFiles: string[] = []
 
 		// TODO(xatier): add darwin support
 		// find ~/.config/{browser} -name Bookmarks
-		if (this.api.environment.platform === 'linux') {
+		switch (this.api.environment.platform) {
+		case 'linux': {
+			const supportedBrowsers = [ 'microsoft-edge-dev', 'chromium', 'google-chrome' ]
 			supportedBrowsers.forEach((browser) => {
 				bookmarkFiles.push(...walk(`${this.api.environment.homedir}/.config/${browser}`, []).filter((f) => f.endsWith('/Bookmarks')))
```

Comment:
question: should we use `joinPath` here as well?

Reply:
sure. but we can fix this later